pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -1,22 +1,192 @@
+    "gnutls" = [ gnutls ];
+    "kmod" = [ kmod ];
+    "libcryptsetup" = [ cryptsetup ];
+    "libcurl" = [ curl ];
+    "libidn" = [ libidn ];
+    "libidn2" = [ libidn2 ];
+    "libiptc" = [ iptables ];
+    "lz4" = [ lz4 ];
+    "microhttpd" = [ libmicrohttpd ];
+    "pam" = [ pam ];
+    "pcre2" = [ pcre2 ];
+    "qrencode" = [ qrencode ];
+    "seccomp" = [ libseccomp ];
+    "selinux" = [ libselinux ];
+    "xkbcommon" = [ libxkbcommon ];
+    "xz" = [ xz ];
+    "zlib" = [ zlib ];
+  };
+
+  featureNativeBuildInputs = {
+    "html" = [ libxslt.bin docbook_xsl docbook_xml_dtd_42 docbook_xml_dtd_45 ];
+    "man" = [ libxslt.bin docbook_xsl docbook_xml_dtd_42 docbook_xml_dtd_45 pythonLxmlEnv ];
+  };
+
+  metaFeatures = [
+    "dns-over-tls" # requires "gnutls" feature as well
+    "importd" # requires "libcurl", "zlib", "xz", and "gcrypt" features also
+    "polkit" # if polkit package is present, there's a build-time version check
+    "remote" # requires "microhttpd" and "libcurl" features also
+    "split-bin"
+    "split-usr"
+  ];
+
+  # Disable all features that have dependencies, then turn back on the features
+  # that NixOS wants.
+  defaultFeatures = lib.genAttrs allFeatures (_: false) //
+    {
+      # Features systemd would try to auto-detect that we want turned on:
+      "acl" = true;
+      "apparmor" = true;
+      "audit" = true;
+      "blkid" = true;
+      "bzip2" = true;
+      "gcrypt" = true;
+      "glib" = true;
+      "gnu-efi" = !stdenv.isAarch32 && !stdenv.isAarch64 && stdenv.hostPlatform.isEfi;
+      "kmod" = true;
+      "libidn2" = true;
+      "libiptc" = true;
+      "lz4" = true;
+      "man" = true;
+      "microhttpd" = true;
+      "pam" = true;
+      "pcre2" = true;
+      "polkit" = true;
+      "remote" = "auto"; # only way to get sd-j-remote without sd-j-upload
+      "seccomp" = withLibseccomp;
+      "selinux" = withSelinux;
+      "split-bin" = true;
+      "xz" = true;
+      "zlib" = true;
+
+      # Other flags:
+      "tests" = false; # TODO: fails a bunch of tests
+      "hostnamed" = true;
+      "networkd" = true;
+      "sysusers" = false;
+      "timedated" = true;
+      "timesyncd" = true;
+      "firstboot" = false;
+      "localed" = true;
+      "resolve" = true;
+      "quotacheck" = false;
+      "ldconfig" = false;
+      "smack" = true;
+
+      # NixOS-specific settings, overridable if desired:
+      "tty-gid" = 3; # tty in NixOS has gid 3
+      "system-uid-max" = 499; #TODO: debug why awking around in /etc/login.defs doesn't work
+      "system-gid-max" = 499;
+      "ntp-servers" = "0.nixos.pool.ntp.org 1.nixos.pool.ntp.org 2.nixos.pool.ntp.org 3.nixos.pool.ntp.org";
+      # "time-epoch" = 1;
+
+      # NixOS doesn't need SysV init compatibility.
+      "sysvinit-path" = "";
+      "sysvrcnd-path" = "";
+
+      # Needed if the gnu-efi feature is enabled; ignored otherwise.
+      "efi-includedir" = "${toString gnu-efi}/include/efi";
```

Comment:
Why do you need `toString`?

Reply:
I think you're right that this doesn't need `toString`, but I copied those settings directly from the existing code and I didn't want to change anything like that. Those `toString` calls are still in the current version of the package, even.